### PR TITLE
chore: update related articles section

### PR DESCRIPTION
**Pull request by manika-signoz**
## Summary

Updates the article layout so the related-articles block shows three items instead of two, and moves the newsletter subscription section to appear after related articles (instead of before) and styling changes to related articles section.

Closes https://github.com/SigNoz/engineering-pod/issues/4727 

## Changes

- Increase related articles list from 2 to 3 (`slice(0, 3)`).
- Relocate the newsletter block below the related-articles section in the main column.

## Type of Change

- [ ] **Docs** – Changes to `data/docs/**`
- [ ] **Blog** – Changes to `data/blog/**`
- [x] **Site Code** – Changes to `app/**`, `components/**`, `hooks/**`, `utils/**`, config, etc.
- [ ] **Redirects** – Renamed/moved docs or updated `next.config.js` redirects
- [ ] **Dependencies / CI / Scripts**
- [ ] **Other** – Explain below

## Impact

- [ ] Documentation only
- [x] UI changes
- [ ] Developer workflow
- [ ] Performance impact
- [ ] Breaking change

## Context & Screenshots
Before
Screenshot 2026-05-05 at 12 53 19
Screenshot 2026-05-05 at 12 53 52

After
Screenshot 2026-05-05 at 12 54 17

Screenshot 2026-05-05 at 12 52 41

### General

- [x] Branch is up to date with `main`
- [x] PR title follows conventional format (`type: description`)
- [x] Self-reviewed the code
- [ ] Comments added for complex logic

## Testing

- [ ] Tested locally
- [ ] Edge cases considered
- [ ] Existing functionality verified
- [ ] New tests added (if applicable)

Steps to test:

1. Open any article/blog page that uses `ArticleLayout` with related articles enabled.
2. Confirm up to three related article links render when enough related content exists.
3. Confirm the newsletter block appears below the related-articles section.

## Rollout / Deployment Notes

None.

---

**Note:** Submit as Draft by default. Mark "Ready for review" when checks pass and content is ready.

### CODE REVIEW

File: layouts/ArticleLayout.tsx [hunk 1]
```diff
@@ -284,60 +284,57 @@ export default function ArticleLayout({
...
+                        target="_blank"
+                        clickType="Nav Click"
+                        clickName="Related Article Link"
+                        clickText={article.title}
+                        clickLocation={`${contentType} Related Articles`}
+                        className="group flex flex-col justify-between rounded-xl border border-signoz_ink-300 bg-signoz_ink-400/50 p-5 transition-all duration-200 hover:border-signoz_robin-500/60 hover:bg-signoz_ink-400"
+                      >
+                        <div>
+                          <p className="mb-3 text-[11px] font-medium uppercase tracking-widest text-signoz_robin-400/70">
+                            {new Date(article.publishedOn || article.date).toLocaleDateString(
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply layouts/ArticleLayout.tsx:314]: this is needed as fallback

File: layouts/ArticleLayout.tsx [hunk 2]
```diff
@@ -284,60 +284,57 @@ export default function ArticleLayout({
...
+                    {relatedArticles.slice(0, 3).map((article, index) => (
+                      <TrackingLink
+                        key={index}
+                        href={article.url}
+                        target="_blank"
+                        clickType="Nav Click"
+                        clickName="Related Article Link"
+                        clickText={article.title}
+                        clickLocation={`${contentType} Related Articles`}
+                        className="group flex flex-col justify-between rounded-xl border border-signoz_ink-300 bg-signoz_ink-400/50 p-5 transition-all duration-200 hover:border-signoz_robin-500/60 hover:bg-signoz_ink-400"
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply layouts/ArticleLayout.tsx:310]: transition-all is fine since transition is needed on all for this element, will narrow if use case arises

File: layouts/ArticleLayout.tsx [hunk 3]
```diff
@@ -284,60 +284,57 @@ export default function ArticleLayout({
...
-                    </div>
+                <div className="mt-12 border-t border-signoz_ink-300 pt-10">
+                  <div className="mb-6">
+                    <p className="mb-1 text-xs font-medium uppercase tracking-[0.2em] text-signoz_robin-400">
+                      Keep Reading
+                    </p>
+                    <h2 className="text-xl font-semibold text-white">Related Articles</h2>
+                  </div>
+                  <div className="grid gap-4 sm:grid-cols-2 lg:grid-cols-3">
+                    {relatedArticles.slice(0, 3).map((article, index) => (
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply layouts/ArticleLayout.tsx]: done

**Review by YounixM — APPROVED** [review]
